jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +705,290 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
                 self.subset_definitions[index][i]['value'] = new_value
                 break
 
-    def _import_spectral_regions(self, spec_region, mode=NewMode, viewer=None):
+    def import_region(self, region, combination_mode=None, max_num_regions=None,
+                      refdata_label=None, return_bad_regions=False, **kwargs):
         """
-        Method for importing a SpectralRegion object or list of SpectralRegion objects.
+        Method for creating subsets from regions or region files.
 
         Parameters
         ----------
-        spec_region : ~`specutils.SpectralRegion` object or list
-            The object that contains bounds for the spectral region or regions
-        mode : AndMode, AndNotMode, OrMode, ReplaceMode, XorMode, NewMode, or list
-            By default this is set to NewMode which creates a new subset per subregion.
-            OrMode will create a composite subset with each subregion corresponding
-            to a subregion of a single subset.
+        region : region, list of region objects, or str
+            A region object can be one of the following:
+
+            * Astropy ``regions`` object
+            * ``photutils`` apertures (limited support until ``photutils``
+              fully supports ``regions``)
+            * specutils ``SpectralRegion`` object
+
+            A string which represents a ``regions`` or ``SpectralRegion`` file.
+            If given as a list, it can only contain spectral or non-spectral regions, not both.
+
+        combination_mode : list, str, or `None`
+            The way that regions are created or combined. If a list, then it must be the
+            same length as regions. If `None`, then it will follow the default glue
+            functionality for subset creation.
+
+        max_num_regions : int or `None`
+            Maximum number of regions to load, starting from top of the list.
+            Default is to load everything.  If you are providing a large file/list
+            input for ``region``, it is recommended
+
+        refdata_label : str or `None`
```

Comment:
I wonder if we could just do `data_label` - but we can revise this anytime before making public, so can ignore for now.